migrate_legacy_captures.js

```diff
@@ -0,0 +1,218 @@
+      .first();
+  }
+  if (!planter) {
+    throw Error(
+      `No planter record found for tree entry with id ${legacyTree.id}`
+    );
+  }
+  let growerAccount = await transaction("treetracker.grower_account")
+    .where("email", planter.email)
+    .orWhere("phone", planter.phone)
+    .first();
+  if (!growerAccount) {
+    console.log(
+      `GrowerAccount not found for planter_identifier ${legacyTree.planter_identifier}`
+    );
+    console.log(
+      `Populating grower account for ${legacyTree.planter_identifier}`
+    );
+    // status, first_registration_at aren't available in the source `public.planter` table
+    // Is it ok, if the above attributes take the db defaults?
+    growerAccount = await transaction("treetracker.grower_account").insert(
+      {
+        name: planter.first_name + " " + planter.last_name,
+        email: planter.email,
+        phone: planter.phone,
+        image_url: planter.image_url ?? legacyTree.planter_photo_url, //should this be public.trees.planter_photo_url?
+        image_rotation: planter.image_rotation ?? 0,
+        organization_id: planter.organization_id,
+        wallet_id: generate_uuid_v4(), // TODO: temporary workaround, needs to be a specific uuid and not a newly generated one.
+        wallet: "Temporary placeholder",
+        first_registration_at: new Date(),
+      },
+      [
+        "id",
+        "wallet_id",
+        "name",
+        "email",
+        "phone",
+        "organization_id",
+        "image_url",
+        "image_rotation",
+      ]
+    );
+  }
+  console.log(`GrowerAccount exists: ${JSON.stringify(growerAccount)}`);
+  return growerAccount;
+}
+
+async function migrate() {
+  try {
+    const base_query_string = `select t.* from public.trees t left join treetracker.capture c on t.id = c.reference_id
+                                where t.active=true and t.approved=true and t.image_url != null and
+                                c.reference_id is null`;
+    const rowCountResult = await knex.select(
+      knex.raw(`count(1) from (${base_query_string}) as src`)
+    );
+    console.log(`Migrating ${+rowCountResult[0].count} records`);
+
+    const bar = new ProgressBar("Migrating [:bar] :percent :etas", {
+      width: 20,
+      total: +rowCountResult[0].count,
+    });
+
+    const attributesFormatter = (attributes) => {
+      if (attributes.length <= 0) return null;
+      const attributes_in_json_format = { entries: [] };
+      for (let entry of attributes) {
+        attributes_in_json_format.entries.push(entry);
+      }
+      return attributes_in_json_format;
+    };
+
+    const createCapture = (
+      {
+        id,
+        time_created,
+        planter_photo_url,
+        gps_accuracy,
+        image_url,
+        lat,
+        lon,
+        morphology,
+        age,
+        note,
+        time_updated,
+      },
+      treeAttributes,
+      growerAccount
+    ) => {
+      return Object.freeze({
+        reference_id: id,

```

Comment:
Is it accurate to point `treetracker.capture.reference_id` to `public.trees.id`?

Reply:
yes